File: atcoder/abc135A_harmony_.hs
-- uninhm
-- https://atcoder.jp/contests/abc135/tasks/abc135_a
-- basic math

solve [a, b] = 
	if (a + b) `mod` 2 == 0 then
		show $ (a + b) `div` 2
	else
		"IMPOSSIBLE"

main = interact $ solve . map read . words

File: atcoder/abc135A_harmony.ml
(* vicfred
 * https://atcoder.jp/contests/abc135/tasks/abc135_a
 * math
 * *)
Scanf.scanf "%d %d\n" (fun a b ->
    if (a + b) mod 2 != 0 then "IMPOSSIBLE"
    else string_of_int @@ (a + b) / 2
) |> Printf.printf "%s\n"

